client/cli/src/client_proxy.rs

```diff
@@ -753,21 +753,23 @@ impl ClientProxy {
 
     /// Publish move module
     pub fn publish_module(&mut self, space_delim_strings: &[&str]) -> Result<()> {
-        let module = serde_json::from_slice(&fs::read(space_delim_strings[2])?)?;
-        self.submit_program(space_delim_strings, TransactionPayload::Module(module))
+        let module_bytes = &fs::read(space_delim_strings[2])?;
+        self.submit_program(
+            space_delim_strings,
+            TransactionPayload::Module(Module::new(module_bytes.clone())),
+        )
     }
 
     /// Execute custom script
     pub fn execute_script(&mut self, space_delim_strings: &[&str]) -> Result<()> {
-        let script: Script = serde_json::from_slice(&fs::read(space_delim_strings[2])?)?;
-        let (script_bytes, _) = script.into_inner();
+        let script_bytes = &fs::read(space_delim_strings[2])?;
         let arguments: Vec<_> = space_delim_strings[3..]
             .iter()
             .filter_map(|arg| parse_as_transaction_argument_for_client(arg).ok())
             .collect();
         self.submit_program(
             space_delim_strings,
-            TransactionPayload::Script(Script::new(script_bytes, arguments)),
+            TransactionPayload::Script(Script::new(script_bytes.clone(), arguments)),
```

Comment:
Redundant `clone`?

Reply:
script_bytes is a reference to a Vec<u8> but Script::new expects a non-reference Vec<u8>. If I remove the clone(), the compiler suggests using to_vec() instead. If I dereference script_bytes, I get an error: "cannot move out of `*script_bytes` which is behind a shared reference".  I tried to_vec() and that also works, but it is also making a new copy of the vector. Is there any reason to prefer to_vec() over clone()?

Reply:
Ah, I think if you change `let script_bytes = &fs::read(space_delim_strings[2])?;` to `let script_bytes = fs::read(space_delim_strings[2])?;` (remove `&`) it will work. But not a big deal, in any case.

Reply:
Well, duh. Of course that is a better way to remove the reference! I was just copying from what had been there before, so I appreciate the suggestion.